src/memtypes.py

```diff
@@ -158,10 +159,14 @@ def is_false(self) -> bool:
     def __str__(self):
         if self.is_unconstrained:
             return self.identifier
-        if self.is_const:
-            return hex(self.const_value)
-        val_str = ", ".join([hex(val) for val in self.value])
-        return "{{{}}}".format(val_str)
+        elif self.is_const:
+            if settings.output_replace_var_values or self.identifier == "C":
```

Comment:
This seems opaque. It might not be immediately clear why this is this way to a reader unfamiliar with how const-valued variables are generated from `PUSH` operations.

Perhaps a a comment explaining this would be helpful, or maybe a small utility function function dedicated to this purpose?